### PR TITLE
ci: add sync-param-files

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

add sync setting file to sync `awf/autoware_launch/autoware_launch/config/tier4_*_launch` with  `awf/autoware.universe/launch/tier4_*_launch/config`

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@kenji-miyake 
Sorry to mention you again. :bow: 

I've created sync PR from autoware.universe/launch/tier4_*_launch/config -> autoware_launch/config/tier4_*_launch
How can I verify this sync setting file will work well.

File: .github/sync-universe-param-files.yaml
Comment (kenji-miyake) [inline .github/sync-universe-param-files.yaml:3]: For better visibility, it's good to add sections.

```suggestion
    # Perception
    - source: launch/tier4_perception_launch/config/object_recognition/tracking/multi_object_tracker/data_association_matrix.param.yaml
```

File: .github/workflows/sync-universe-param-files.yaml
Comment (kenji-miyake) [inline .github/workflows/sync-universe-param-files.yaml:9]: It's not limited to only `universe`.

```suggestion
name: sync-param-files

on:
  schedule:
    - cron: 0 0 * * *
  workflow_dispatch:

jobs:
  sync-param-files:
```

File: .github/workflows/sync-universe-param-files.yaml
Comment (kenji-miyake) [inline .github/workflows/sync-universe-param-files.yaml:26]:
```suggestion
          config: .github/sync-param-files.yaml
          pr-branch: sync-param-files
          pr-title: "chore: sync param files"
          pr-commit-message: "chore: sync param files"
```

**Review by kenji-miyake — CHANGES_REQUESTED** [review]

File: .github/sync-universe-param-files.yaml
Comment (takayuki5168) [reply .github/sync-universe-param-files.yaml]: https://github.com/autowarefoundation/autoware_launch/pull/64/commits/a72531e97e0a0d1b1f1fae0a7eb4081cc311abae
DONE

File: .github/workflows/sync-universe-param-files.yaml
Comment (takayuki5168) [reply .github/workflows/sync-universe-param-files.yaml]: https://github.com/autowarefoundation/autoware_launch/pull/64/commits/a72531e97e0a0d1b1f1fae0a7eb4081cc311abae
DONE

File: .github/workflows/sync-universe-param-files.yaml
Comment (takayuki5168) [reply .github/workflows/sync-universe-param-files.yaml]: https://github.com/autowarefoundation/autoware_launch/pull/64/commits/a72531e97e0a0d1b1f1fae0a7eb4081cc311abae
DONE

**Review by kenji-miyake — APPROVED** [review]

**Comment by takayuki5168** [comment]
@kenji-miyake I haven't checked if this PR works, but is it ok to merge this...? (After merged, we can enable this sync PR?)

**Comment by kenji-miyake** [comment]
@takayuki5168 Yes, since this just adds a new workflow, I think it's okay to merge this.
We can test new workflows only after they're merged to the main branch.

**Comment by kenji-miyake** [comment]
But regarding the sync PR that will be generated by this workflow, we should review it.

**Comment by kenji-miyake** [comment]
@takayuki5168 Trigger this workflow to test it.
https://github.com/autowarefoundation/autoware_launch/actions/workflows/sync-param-files.yaml

**Pull request merged**
